pkg/cmd/pulumi/deployment_settings_utils.go

```diff
@@ -0,0 +1,247 @@
+	}
+	return psd.Save(stackDeploymentConfigFile)
+}
+
+type prompts interface {
+	PromptUser(msg string, options []string, defaultOption string,
+		colorization colors.Colorization,
+	) string
+	PromptUserSkippable(yes bool, msg string, options []string, defaultOption string,
+		colorization colors.Colorization,
+	) string
+	PromptUserMultiSkippable(yes bool, msg string, options []string, defaultOptions []string,
+		colorization colors.Colorization,
+	) []string
+	PromptForValue(
+		yes bool, valueType string, defaultValue string, secret bool,
+		isValidFn func(value string) error, opts display.Options,
+	) (string, error)
+	AskForConfirmation(prompt string, color colors.Colorization, defaultValue bool, yes bool) bool
+}
+
+type promptHandlers struct{}
+
+func (promptHandlers) AskForConfirmation(prompt string, color colors.Colorization, defaultValue bool, yes bool) bool {
+	return askForConfirmation(prompt, color, defaultValue, yes)
+}
+
+func (promptHandlers) PromptUser(msg string, options []string, defaultOption string,
+	colorization colors.Colorization,
+) string {
+	return promptUser(msg, options, defaultOption, colorization)
+}
+
+func (promptHandlers) PromptUserSkippable(yes bool, msg string, options []string, defaultOption string,
+	colorization colors.Colorization,
+) string {
+	return promptUserSkippable(yes, msg, options, defaultOption, colorization)
+}
+
+func (promptHandlers) PromptUserMultiSkippable(yes bool, msg string, options []string, defaultOptions []string,
+	colorization colors.Colorization,
+) []string {
+	return promptUserMultiSkippable(yes, msg, options, defaultOptions, colorization)
+}
+
+func (promptHandlers) PromptForValue(
+	yes bool, valueType string, defaultValue string, secret bool,
+	isValidFn func(value string) error, opts display.Options,
+) (string, error) {
+	return promptForValue(yes, valueType, defaultValue, secret, isValidFn, opts)
+}
+
+type repoLookup interface {
+	GetRootDirectory(wd string) (string, error)
+	GetBranchName() string
+	RemoteURL() (string, error)
+	GetRepoRoot() string
+}
+
+func newRepoLookup(wd string) (repoLookup, error) {
+	repo, err := git.PlainOpenWithOptions(wd, &git.PlainOpenOptions{DetectDotGit: true})
+	switch {
+	case errors.Is(err, git.ErrRepositoryNotExists):
+		return &noRepoLookupImpl{}, nil
+	case err != nil:
+		return nil, err
+	}
+
+	worktree, err := repo.Worktree()
+	if err != nil {
+		return nil, err
+	}
+
+	h, err := repo.Head()
+	if err != nil {
+		return nil, err
+	}
+
+	return &repoLookupImpl{
+		RepoRoot: worktree.Filesystem.Root(),
+		Repo:     repo,
+		Head:     h,
+	}, nil
+}
+
+type repoLookupImpl struct {
+	RepoRoot string
+	Repo     *git.Repository
+	Head     *plumbing.Reference
+}
+
+func (r *repoLookupImpl) GetRootDirectory(wd string) (string, error) {
+	dir, err := filepath.Rel(r.RepoRoot, wd)
+	if err != nil {
+		return "", err
+	}
+
+	// we have to convert non linux to use the linux path separator
+	return convertToLinuxPath(dir), err
```

Comment:
I'm not sure I quite understand why this should be converted to a Unix path.  I understand deployments is always using Linux containers, but if I'm reading things correctly, it always points to a *local* directory anyway, not the remote one?  Also the user is prompted for a value for the repository root, which they may or may not enter in the native or Unix name, so converting to the Unix path should be happening *after* that in any case.  Am I missing something here?  

As an aside, all the deployment settings struct could use some more documentation.

Reply:
> it always points to a local directory anyway
yes and that configuration applies when running the deployment, which runs in a unix environment, hence the need to use the appropriate path separator

> so converting to the Unix path should be happening after that in any case
makes sense, my intent was to keep this input consistent across the different interfaces where it is expected to be "unix format". But makes sense to keep it consistent with the environment it is running too, I will change

Reply:
done, I also remove the two "convert path" functions and used filepath's ToSlash and FromSlash implementation